### DISCUSSION TITLE
Adding ESS (Enhanced Signin Security) support to the script

### DISCUSSION THREAD

**Discussion by agpt8** [HotCakeX/Harden-Windows-Security#163] [General]
Hi there,

I noticed that under Win 11 Secure Core PC's section, there is a category that is not yet applied with the Hardening script, namely [Enhanced Signin Security](https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/restore-scpc-config-for-admins-11#enhanced-sign-in-security-ess)

Is this something that should be added explicitly or is it applied in some other way by the script?

I dont see the keys (specifically, SecureBiometrics and SecureFingerprint) in the Registery Editor at the path given in the section above.
Is this something that should be added to the script?

**Comment by HotCakeX** [comment]
Hi,
Microsoft security baselines include it, you can [download it](https://www.microsoft.com/en-us/download/details.aspx?id=55319) and search for `Enhanced Sign-in Security (ESS)` in the included Excel document. Although looks like we don't need to enable it because it's enabled by default according to the Group policy description

<img width="1280" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/c134ea97-227c-42fc-b6af-549ec93cb70f">

<br>

<br>

It was automatically enabled for me. Windows Security -> Device Security

<img width="572" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/56ae71ea-eab7-4aef-aa17-cbc44eb0ca96">

<br>

**Reply by agpt8** [reply]
Interesting! I dont see this ESS section on my end.

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/16762897/11fb7373-0608-44bf-b8c0-932edc3a77ed)

I guess this is only effective for newer Secured Core PCs..? I have added those two keys manually but not sure if they'll help in any way.

**Reply by HotCakeX** [reply]
Oh it should be in the Core Isolation section

<img width="1019" alt="image" src="https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/638aedd0-13ea-4f48-9496-11016806473d">

It's not a requirement for a Secured-core PC though.

ESS has some requirements such as compatible hardware and driver: https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/windows-hello-enhanced-sign-in-security#system-compatibility

**Reply by agpt8** [reply]
The Core Isolation page also does not list this:

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/16762897/f7adffe6-f748-4f6e-a532-77f4ed34aa0f)


I did check the hardware compatibility as per the given link, it seems my workstation does not meet the requirements.

Just one last query though, is there an easy way to know which all PCs and workstations support these? And if someone is building themselves a PC, how can they ensure that the parts they are selecting meet these requirements? Or is it even possible on desktop PCs?

**Reply by HotCakeX** [reply]
AFAIK PCs with computer cases don't support this feature. The webcam or fingerprint scanner should be internal not external to support ESS. So I think only laptops and tablets support this feature.

I'd be looking at business class laptops/tablets on the manufacturers' websites. They're the ones that have things like discrete TPM (dTPM) instead of the virtualized/firmware TPM (fTPM) found on normal consumer devices, etc.

Microsoft Surface products are easy, the business class versions are secured core and ship with all the additional security features that the consumer Surface products don't have.

**Reply by HotCakeX** [reply]
https://support.microsoft.com/en-us/windows/enhanced-sign-in-security-in-windows-936c06ed-4024-4521-921f-faa97925a16e

**Reply by HotCakeX** [reply]
https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/windows-hello-face-authentication

**Reply by agpt8** [reply]
This was really helpful. Thank you! I'll close the discussion.
